src/algebraic_geometry/elliptic_curve/weierstrass.lean

```diff
@@ -316,51 +316,57 @@ end
 
 /-! ### Nonsingularity of Weierstrass curves -/
 
-/-- The partial derivative $W_X(X, Y)$ of $W(X, Y)$ with respect to $X$. -/
+/-- The partial derivative $W_X(X, Y)$ of $W(X, Y)$ with respect to $X$.
+
+TODO: define this in terms of `polynomial.derivative`. -/
 noncomputable def polynomial_X : R[X][X] :=
```

Comment:
Here you haven't incorporated the R[X][Y] change. Even though we can't yet make the infoview distinguish X and Y, it would still be nice to use Y in statements, right? Can you simply copy the content of weierstrass.lean in #18000 to this PR (except the two declarations I added)?